tests/unit/test_pytest_operator.py

```diff
@@ -60,6 +60,24 @@ async def test_without_tox(request, ops_test):
     result.assert_outcomes(passed=1)
 
 
+@patch.object(plugin.OpsTest, "default_model_name", "user/model")
+@patch.object(plugin.OpsTest, "_setup_model", AsyncMock())
+@patch.object(plugin.OpsTest, "_cleanup_models", AsyncMock())
+def test_tmp_path_with_nonpath_chars(pytester):
+    pytester.makepyfile(
+        f"""
+        import os
+        from pathlib import Path
+
+        os.environ.update(**{ENV})
+        async def test_with_tox(ops_test):
+            assert ops_test.tmp_path.name == "user_model0"
+        """
+    )
+    result = pytester.runpytest()
+    result.assert_outcomes(passed=1)
+
+
```

Comment:
So, that test was kinda confusing to me.  Can we do this:

```suggestion
async def test_tmp_path_with_nonpath_chars(tmp_path_factory, setup_request):
    setup_request.module.__name__ = "user/model"
    ops_test = plugin.OpsTest(setup_request, tmp_path_factory)
    assert ops_test.tmp_path.name.startswith("user_model")
```